processing/data_collection/gazette/pipelines.py

```diff
@@ -22,6 +20,30 @@ def process_item(self, item, spider):
         return item
 
 
+class ExtraIssuePipeline:
+    """
+    Try to identify whether or not a given Gazette is an extraordinary
+    issue based on the item's source_text, and a spider-defined pattern
+    """
+
+    def process_item(self, item, spider):
+        # Avoid touching an item that was manually defined.
+        if "is_extra_edition" in item:
+            return item
+
+        # No source_text, nothing to do
+        if item["source_text"] is None:
+            return item
+
+        # If the spider has not defined a pattern, go away
+        if not hasattr(spider, "EXTRA_ISSUE_PATTERN"):
```

Comment:
I'd put this check before the item content

Reply:
Done.